src/NovelRT/Atom.cpp

```diff
@@ -5,45 +5,86 @@
 
 namespace NovelRT
 {
-    Atom Atom::getNextEventHandlerId() noexcept
+    Atom Atom::GetNextEventHandlerId() noexcept
     {
         static std::atomic_uintptr_t _nextEventHandlerId(0);
         auto value = ++_nextEventHandlerId;
         return Atom(value);
     }
 
-    Atom Atom::getNextFontSetId() noexcept
+    Atom Atom::GetNextFontSetId() noexcept
     {
         static std::atomic_uintptr_t _nextFontSetId(0);
         auto value = ++_nextFontSetId;
         return Atom(value);
     }
 
-    Atom Atom::getNextTextureId() noexcept
+    Atom Atom::GetNextTextureId() noexcept
     {
         static std::atomic_uintptr_t _nextTextureId(0);
         auto value = ++_nextTextureId;
         return Atom(value);
     }
 
-    Atom Atom::getNextComponentTypeId() noexcept
+    Atom Atom::GetNextComponentTypeId() noexcept
     {
         static std::atomic_uintptr_t _nextComponentTypeId(0);
         auto value = ++_nextComponentTypeId;
         return Atom(value);
     }
 
-    Atom Atom::getNextEntityId() noexcept
+    Atom Atom::GetNextEntityId() noexcept
     {
         static std::atomic_uintptr_t _nextEntityId(0);
         auto value = ++_nextEntityId;
+
+        if (value == std::numeric_limits<uintptr_t>::max())
```

Comment:
Surely if this branch is reached we have a problem anyway? uintptr_t is 2^64 on modern 64-bit builds. That's 18 quintillion. I'm certain any system we'd be running on would run out of memory far before we reached that number, as 18 quintillion equates to roughly 18 exabytes by itself.

Reply:
Particle systems have entered the chat (in reality I have no idea and also didn't want to find out).

Reply:
I imagine a *good* particle system would allocate a pool of entities to use for particle effects and then re-use them when available? Something like this really bad pseudocode:
```
std::queue<EntityId> pool;

void run_particle_system()
{
    if (pool.empty())
    {
        pool.push(new_particle());
    }
    else
    {
        auto entity = pool.front();
        do_stuff(entity);
        pool.pop();
    }
}

void do_stuff(EntityId entity)
{
    // magic
    pool.push(entity);
}
```

Reply:
Yeah I know, and this branch will probably never ever get called, I just really didn't like leaving the possibility lurking around.